tests/test_loss.py

```diff
@@ -107,8 +108,10 @@ def test_unbinned_simultaneous_nll():
 
 
 @pytest.mark.flaky(3)
-@pytest.mark.parametrize('weights', [None, np.random.normal(loc=1., scale=0.2, size=test_values_np.shape[0])])
-def test_unbinned_nll(weights):
+@pytest.mark.parametrize('weights, sigma', [(None, [mu_constr[1], sigma_constr[1]]), (None, covariance),
+                         (np.random.normal(loc=1., scale=0.2, size=test_values_np.shape[0]),
+                         [mu_constr[1], sigma_constr[1]])])
+def test_unbinned_nll(weights, sigma):
```

Comment:
(sorry for my slow reaction recently, been quite busy)
Can you do two changes? I think then we're good

- separate the parametrize? They can simply be stacked. So that there is a weight parameterization and a gaussian constraint parametrisation.
- can you make the constraints callables? (and in the code just call them?) This allows to also have tensors in the test, since they cannot be build yet outside the function (e.g. in the parametrize) and add a tensor as test? (can be simply: lambda: tf.constant(mu_constr[0])

(not for _now_, this works of course, but for future that we don't break things)

Reply:
I don't understand why making the constraints callable?

Reply:
@marinang according to my email, you posted something, asking why the callable is needed. I don't see it on github. Weird...
Anyway, the reason is that the parametrization of the test happens at collection time, so an arbitrary time _before_ the test is executed. Since, due to the setup_function, the graph is reset before each test (to have clean tests). Since a tensor is added to the graph when the method is called (e.g. `tf.constant` this would add the constant to the graph at collection time. This gets reset and at the test runtime, this would point to an object from a different (because reset) graph. With the callable, we make sure that the actual op gets added to the right graph.

On the parametrization: you can remove most of them, it is the product of combinations, a matrix of the possible combinations.

so:
```
parametrized("weights", [None, something])
parametrized("other", [a, b, c])
```
gives 2 x 3 combinations of tests that are run.

The CI is failing, I think due to an indentation error. Typo somewhere?

Reply:
Ok great I just discovered very recently the parametrize option of pytest, nice feature. For the failure of “pr” Travis build, there is a lot of red everywhere and it’s difficult for me to catch where the error comes from (also not easy to read the log on a phone).

Reply:
I agree, I don't see it either. Anyway, I fix the pytest parametrization and the bug